README.md

```diff
@@ -51,6 +51,14 @@ core distribution of the Collector. Typically, these contributions are vendor
 specific receivers/exporters and/or components that are only
 useful to a relatively small number of users.
 
+## Creating your own OpenTelemetry Collector distribution
+
+A guide on how to create your own distribution is available in this blog post: 
+["Building your own OpenTelemetry Collector distribution"](https://medium.com/p/42337e994b63)
+
+If you are building your own distribution, the [OpenTelemetry Collector Builder](https://github.com/observatorium/opentelemetry-collector-builder) 
+might be a good starting point.
+
```

Comment:
Do you think we can add a point that if you use a module that has an internal dependency you may need to fix versions for it too? We ran into that here

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1217

Reply:
How often do you think people would run into this problem? In my opinion, the decision to depend on `contrib` is a design decision by the component: either create the `internal` package inside its own Go module, or create it elsewhere (`contrib`) and make consumers pull this transitive dependency.

We can add a word about it, but I think it's just how Go applications usually work anyway (or I'm failing to see the actual problem here).

Reply:
I'm not a Go expert, but I think it is supposed to usually work OK and previously did here too. This repo has a special issue related to dependabot though that makes it not idiomatic anymore

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1219#issuecomment-705303444

So without fixing that, I think it's worth calling out here so that someone doing a similar redistribution does not run into the cryptic issue for better user experience.

Reply:
How about we wait for another module to get into the same problem to see how frequent this happens? I would try to avoid mentioning edge cases in a general doc, as it has the potential to confuse readers who would not typically run into such problems.